Inventory refactored. Window actions moved to plugins. Chests. Furnaces.

mcsakoff opened the pull request:
- Bravo.inventory refactoring.
- Window actions moved to plugins. (Do not forget to update you .ini file)
- Chests implemented.
- Initial support for furnaces. Furnace window works as expected but processing is not implemented yet.
- Support for shared containers. Two or more players can open and use chest/furnace at the same time.
- Some fixes.
- Documentation and unit tests are updated.

mcsakoff committed:
Fix notification about changes to player's equipment.
Code optimization.

mcsakoff committed:
Initial support for chests

mcsakoff committed:
Inventory refactoring:

- Separate such entities as ``window``, ``inventory`` and ``slots set``.
- Initial Chests support (large chest is not supported yet).
- Make serialization of inventory objects sane (no encode/decode rings
  any more).
- Move some method from protocol into inventory window class.

mcsakoff committed:
Split bravo.inventory module to sub-modules

mcsakoff committed:
Implement shift-click on crafting slot

mcsakoff committed:
Convert new files to UNIX format

mcsakoff committed:
Some inventory window basics explained

mcsakoff committed:
Shared containers support. Chest implementation updated.

mcsakoff committed:
- Initial Furnace support
- Fix saving chests content on server shutdown
- Some speed optimizations in bravo.inventory code

mcsakoff committed:
- Furnace window is implemented
- More inventory optimizations and fixes
- Create furnace tile when furnace block is created
- Documentation update

mcsakoff committed:
Merge remote-tracking branch 'upstream/master'

mcsakoff committed:
Merge branch 'chest'

mcsakoff committed:
Cleanup

mcsakoff committed:
Large chests implementation

mcsakoff committed:
Convert bravo/tests/inventory/test_slots.py to UNIX format

mcsakoff committed:
- Move all window actions to plugins
- Implement correct checking for large chests

MostAwesomeDude commented:
I'm not merging this exactly as-is, because I'm not going to merge spurious merge commits. Also, holy crap this is a lot of code. Punting to an unstable branch, chunk-rebased, until I can review this further. I'm going to rebase away the merge commits, and probably also fold the dos2unix commits.

mcsakoff committed:
Fix building large chest

mcsakoff committed:
- Support for unstackable items (not all iteam are defined yet)
- Notchian logic for item collecting to inventory
- Partial item collect

mcsakoff committed:
- Fix right click out of window with item selected
- Add test cases for shift-click on crafting area when inventory is full or almost full

mcsakoff committed:
Drop items from chests and furnaces when the block is removed

mcsakoff committed:
Remove tile when its block is removed

MostAwesomeDude commented:
Merged all of this on chest-rebased.
